packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -206,16 +206,27 @@ export class DefaultCrudRepository<T extends Entity, ID>
     return this.toEntity(model);
   }
 
-  async findById(id: ID, filter?: Filter, options?: Options): Promise<T> {
+  async findById(
+    id: ID,
+    filter?: Filter,
+    options?: Options,
+  ): Promise<T | undefined> {
     const model = await ensurePromise(
       this.modelClass.findById(id, filter, options),
     );
     if (!model) {
-      throw new Error(`no ${this.modelClass.name} found with id "${id}"`);
+      // entity instance not found
+      return undefined;
     }
     return this.toEntity(model);
   }
 
+  async getById(id: ID, filter?: Filter, options?: Options): Promise<T> {
```

Comment:
While I fully understand the motivation to add `getById`, I do have a few concerns:

- `getById` is not a typical method name for CRUD as `get` is usually for accessors or REST `GET`.
- Adding a method with similar function as `findById` might bite us in the long run as we see in LB 3.x. 

I suggest that we add a `helper` method such as `assertFound(entityOrUndefined: Promise<T | undefined>): Promise<T>` instead. 

Case 1: We're sure `c01` exists.
```ts
const customer = await customerRepo.findById('c01');
console.log(customer!.name);
```

Case 2: Convert `undefined` to NotFoundError
```
const customer = await assertFound(customerRepo.findById('c01'));
console.log(customer.name);
```
Case 3: Check the result
```
const customer = await customerRepo.findById('c01');
if (customer) console.log(customer.name);
else throw new Error(...);
```

Reply:
The `assertFound()` helper is concise. I like that.

My concern (which I also raised in my review) is that sleepy developers might simply forget to use the assertion (or the correct method call) and so end up with runtime errors about undefined customers.

So I guess my preference would be to make `findById()` _throw by default_, and have users opt-in to the allow-undefined behaviour. That would make for stricter/tighter code. But if you have rejected that path for sure, then ok never mind me!

One other thing: It might be difficult to translate `assertFound()` over to other functions like `updateById()` and `deleteById()`, if we want to give them a similar and consistent mechanism.